### PR TITLE
IPU changes

### PR DESCRIPTION
**Pull request by turtleli**
Changes:
 - Implements dithering
 - Implements pack command

### CODE REVIEW

**Comment by PSI-Rockin** [comment]
Was the PACK command tested? I know Klonoa uses it, but you'd need a memcard save to easily test it.

**Comment by turtleli** [comment]
PR is partially tested. I made use of the memcard PR.

Klonoa 2 flames:
image

The other game I wanted to test that uses the PACK command (Hisshou pachinko vol 10, requires EE timing hack on PCSX2, RGB16 output + dithering) seems to hang before displaying the opening FMV.

File: src/core/ee/ipu/ipu.cpp [hunk 1]
```diff
@@ -1014,6 +1041,113 @@ bool ImageProcessingUnit::process_CSC()
     }
 }
 
+bool ImageProcessingUnit::process_PACK()
+{
+    while (true)
+    {
+        switch (pack.state)
+        {
+        case PACK_STATE::BEGIN:
```
Comment (PSI-Rockin) [inline src/core/ee/ipu/ipu.cpp:1050]: Add an extra layer of indentation to all case statements in a switch block.

Comment (turtleli) [reply src/core/ee/ipu/ipu.cpp]: Indented.

**Pull request merged**